rollup-node.md

```diff
@@ -0,0 +1,285 @@
+(*[deposits]*).
+
+[L1 attributes transaction]: /glossary.md#l1-attributes-transaction
+
+### From L2 derivation inputs to payload attributes
+
+[payload-attr]: #From-L2-derivation-inputs-to-payload-attributes
+[`PayloadAttributesOPV1`]: #From-L2-derivation-inputs-to-payload-attributes
+
+The rollup reads the following [L2 derivation inputs] for each L1 block:
+
+- L1 block attributes
+   - block number
+   - timestamp
+   - basefee
+- [deposits]
+
+A deposit is an L2 transaction that has been submitted on L1, via a transaction
+sent to the [L2 deposit feed contract][deposit-feed].
+
+While deposits are notably (but not only) used to "deposit" (bridge) ETH and
+tokens to L2, the word *deposit* should be understood as "a transaction
+*deposited* to L2".
+
+The L1 attributes are read from the L1 block header, while deposits are read
+from the block's log entries. Refer to the [**L2 deposit feed contract
+specification**][deposit-feed] for details on how deposits are encoded as
+log entries.
+
+[deposit-feed-spec]: TODO
+
+> **TODO LINK** deposit feed contract specification
+
+From these L2 derivation inputs, the rollup node constructs an expanded version
+of the [Engine API PayloadAttributesV1 object][PayloadAttributesV1]:
+
+[PayloadAttributesV1]: https://github.com/ethereum/execution-apis/blob/main/src/engine/specification.md#payloadattributesv1
+
+```
+PayloadAttributesOPV1: {
+	timestamp: QUANTITY
+	random: DATA (32 bytes)
+	suggestedFeeRecipient: DATA (20 bytes)
+	transactions: array of DATA
+}
+```
+
+The type notation used here refers to the [HEX value encoding] used by the
+[Ethereum JSON-RPC API specification][JSON-RPC-API], as this structure will need to
+be sent over JSON-RPC. `array` refers to a JSON array.
+
+[HEX value encoding]: https://eth.wiki/json-rpc/API#hex-value-encoding
+[JSON-RPC-API]: https://eth.wiki/json-rpc/API
+
+The object properties must be set as follows:
+
+- `timestamp` is set to the current [unix time] (number of elapsed seconds since
+  00:00:00 UTC on 1 January 1970), rounded to the closest multiple of 2 seconds.
+- `random` is currently set to all zero bytes. After [the merge][merge], the randomness
+  value exposed for the L1 block will be copied in this property.
+- `suggestedFeeRecipient` is set to an address where the sequencer would like to
+  direct the fees
+- `transactions` is an array of transactions, RLP-encoded in the [EIP-2718]
+  format (i.e. as a sequence of a single byte and a byte array), of the allowed
+  types. We specify how to select the transactions below.
+
+[unix type]: https://en.wikipedia.org/wiki/Unix_time
+[merge]: https://ethereum.org/en/eth2/merge/
+[EIP-2718]: https://eips.ethereum.org/EIPS/eip-2718
+
+> For details on RLP encoding, refer to the yellow paper, or look at [this
+> hyperlinked implementation][encode-tx].
+
+[encode-tx]: https://github.com/norswap/nanoeth/blob/cc5d94a349c90627024f3cd629a2d830008fec72/src/com/norswap/nanoeth/transactions/Transaction.java#L84-L130
```

Comment:
EIP-2718 defines all we need, we should not leak the details of other transaction types into this specification, or we will have to keep updating it with every L1 change. Also, like the other review comment, we should not rely on external implementations in this specification.

Reply:
Mmh, that is an interesting perspective indeed, I think I agree.

Regarding the impl link, the point is again to be informative, but your point is well-taken. Maybe we should try to mark it more explicitly? I really believe in the value-add here, the encoding scheme is dead simple, but it's hard to piece together from the yellowpaper (and actually the EIPs were most of them are specified & which I don't link here).